api/diagnostics.go

```diff
@@ -514,42 +523,53 @@ func (j *DiagnosticsJob) getHTTPAddToZip(node dcos.Node, endpoints map[string]st
 			}
 
 		default:
-			logrus.Debugf("GET %s", fullURL)
+			logrus.Debugf("GET %s%s", node.IP, httpEndpoint)
 		}
 
-		j.Status = "GET " + fullURL
-		updateSummaryReport("START "+j.Status, node, "", summaryReport)
-		resp, err := get(client, fullURL)
-		if err != nil {
-			j.Errors = append(j.Errors, err.Error())
-			logrus.WithError(err).WithField("URL", fullURL).Error("Could not get from url")
-			updateSummaryReport(err.Error(), node, err.Error(), summaryErrorsReport)
-			j.JobProgressPercentage += percentPerURL
-			continue
-		}
-
-		if resp.Header.Get("Content-Encoding") == "gzip" {
-			fileName += ".gz"
+		status := "GET " + node.IP + httpEndpoint
+		updateSummaryReport("START "+status, node, "", summaryReport)
+		e := j.getDataToZip(node, httpEndpoint, j.client, fileName, zipWriter)
+		updateSummaryReport("STOP "+status, node, "", summaryReport)
+		j.setStatus(status)
+		if e != nil {
+			j.logError(e, node, summaryErrorsReport)
 		}
+		j.incJobProgressPercentage(percentPerURL)
+	}
+	return nil
+}
 
-		// put all logs in a `ip_role` folder
-		zipFile, err := zipWriter.Create(filepath.Join(node.IP+"_"+node.Role, fileName))
-		if err != nil {
-			resp.Body.Close()
-			j.Errors = append(j.Errors, err.Error())
-			logrus.Errorf("Could not add %s to a zip archive: %s", fileName, err)
-			updateSummaryReport(fmt.Sprintf("could not add a file %s to a zip", fileName), node, err.Error(), summaryErrorsReport)
-			j.JobProgressPercentage += percentPerURL
-			continue
-		}
-		io.Copy(zipFile, resp.Body)
-		resp.Body.Close()
-		updateSummaryReport("STOP "+j.Status, node, "", summaryReport)
-		j.JobProgressPercentage += percentPerURL
+func (j *DiagnosticsJob) getDataToZip(node dcos.Node, httpEndpoint string, client *http.Client, fileName string, zipWriter *zip.Writer) error {
+	fullURL, err := util.UseTLSScheme("http://"+node.IP+httpEndpoint, j.Cfg.FlagForceTLS)
+	if err != nil {
+		e := fmt.Errorf("could not read force-tls flag: %s", err)
+		return e
+	}
+	resp, err := get(client, fullURL)
```

Comment:
Since you now set `client` on `*DiagnosticJob` you can reuse it here and don't need to pass it around